packages/manager/apps/web/client/app/domain-operation/update/domain-operation-update.controller.js

```diff
@@ -192,115 +196,97 @@ angular.module('App').controller(
-                    err,
-                    ALERTER_ID,
-                  ),
-                );
-            case 'cancel':
-              return this.domainOperationService[
-                `cancel${this.constructor.capitalizeFirstLetter(
-                  this.type,
-                )}Operation`
-              ](this.operation.id)
-                .then(() =>
-                  this.Alerter.success(
-                    this.$translate.instant('domain_operations_cancel_success'),
-                    ALERTER_ID,
-                  ),
-                )
-                .catch((err) =>
-                  this.Alerter.alertFromSWS(
-                    this.$translate.instant('domain_operations_cancel_error'),
-                    err,
-                    ALERTER_ID,
-                  ),
-                );
-            case 'accelerate':
-              return this.domainOperationService[
-                `accelerate${this.constructor.capitalizeFirstLetter(
-                  this.type,
-                )}Operation`
-              ](this.operation.id)
-                .then(() =>
-                  this.Alerter.success(
-                    this.$translate.instant(
-                      'domain_operations_accelerate_success',
-                    ),
-                    ALERTER_ID,
-                  ),
-                )
-                .catch((err) =>
-                  this.Alerter.alertFromSWS(
-                    this.$translate.instant(
-                      'domain_operations_accelerate_error',
-                    ),
-                    err,
-                    ALERTER_ID,
-                  ),
-                );
-            default:
-              this.Alerter.success(
-                this.$translate.instant('domain_tab_OPERATION_update_success'),
-                ALERTER_ID,
+      for (const arg of this.args) {
+        try {
+          if (arg.type === '/me/document') {
+            if (this.files[arg.key]) {
+              await this.WucUser.uploadFile(arg.key, this.files[arg.key]).then(
+                (documentId) => {
+                  this.domainOperationService.updateOperation({
+                    id: this.operation.id,
+                    key: arg.key,
+                    data: { value: documentId },
+                  });
+                },
               );
-              return null;
+            }
+          } else if (!arg.readOnly) {
+            await this.domainOperationService.updateOperation({
+              id: this.operation.id,
+              key: arg.key,
+              data: { value: arg.value },
+            });
           }
-        })
-        .catch((err) =>
-          this.Alerter.alertFromSWS(
+        } catch (error) {
+          this.displayAlert(
             this.$translate.instant('domain_tab_OPERATION_update_error'),
-            err,
+            ALERTER_ID,
+            error,
+          );
+        }
+      }
+
+      switch (this.todoOperation) {
```

Comment:
nitpick: switch case could be replaced here by dictionnary and compact the code with one call to this.processOperation instead